File: backend/components/metiers/administration_publique/utils/views/api/api_views.test.js
// api_views.test.js – Tests unitaires et conformité API views threed (JS)
const request = require('supertest');
const express = require('express');
const apiRouter = require('./api_views');

describe('API Threed', () => {
  const app = express();
  app.use(express.json());
  app.use('/api', apiRouter);

  it('doit rendre une vue 3D via API', async () => {
    const res = await request(app)
      .post('/api/threed/render')
      .send({ nom: 'Test', statut: 'actif', details: 'ok' });
    expect(res.statusCode).toBe(200);
    expect(res.body.nom).toBe('Test');
    expect(res.body.statut).toBe('actif');
  });
});

File: backend/components/metiers/administration_publique/utils/views/api/api_views.test.py
"""
api_views.test.py – Tests unitaires et conformité API views threed (Python)
"""
from fastapi.testclient import TestClient
from .api_views import router
from fastapi import FastAPI

app = FastAPI()
app.include_router(router)
client = TestClient(app)

def test_render_threed_api():
    response = client.post("/threed/render", json={"nom": "Test", "statut": "actif", "details": "ok"})
    assert response.status_code == 200
    data = response.json()
    assert data["nom"] == "Test"
    assert data["statut"] == "actif"
